src/Features/CSharp/Portable/SignatureHelp/ConstructorInitializerSignatureHelpProvider.cs

```diff
@@ -98,28 +98,30 @@ private static bool IsArgumentListToken(ConstructorInitializerSyntax expression,
                 return null;
             }
 
+            // guess the best candidate if needed and determine parameter index
+            var currentSymbol = semanticModel.GetSymbolInfo(constructorInitializer, cancellationToken).Symbol as IMethodSymbol;
+            var semanticFactsService = document.GetRequiredLanguageService<ISemanticFactsService>();
+            var arguments = constructorInitializer.ArgumentList.Arguments;
+            int parameterIndex;
+            if (currentSymbol is null)
+            {
+                (currentSymbol, parameterIndex) = GuessCurrentSymbolAndParameter(arguments, accessibleConstructors, position, semanticModel, semanticFactsService);
+            }
+            else
+            {
+                // The compiler told us the correct overload, but we need to find out the parameter to highlight given cursor position
+                _ = FindParameterIndexIfCompatibleMethod(arguments, currentSymbol, position, semanticModel, semanticFactsService, out parameterIndex);
+            }
```

Comment:
extract all of this into helper method that returns currentSymbol + parameterindex :)